### ISSUE TITLE
[Change Request]: Include the config options inside of the module itself

### ISSUE THREAD

**Issue by SpadesNeil** [HakonRydland/A3-HR-Garage#10]
### Is the requested change related to a problem?

So right now, when this module is looked at in Eden editor, this is all we see.
![image](https://user-images.githubusercontent.com/6484985/195467906-876f7a62-3c3e-4d8b-a8b5-5c9e1f635893.png)

This is all just the bare bones, and there's not even any module info explaining what it does and how it works. This is not very intuitive.

### Describe the solution you'd like.

Here is an example from a different mod of how the module includes some of its parameters within the module itself.
![image](https://user-images.githubusercontent.com/6484985/195468573-4a43e5f3-5f5d-43a7-a813-2bc0f8463a83.png)

If you could do something similar for this mod, things like the size of the garage, or a blacklist of certain vehicle classnames, a whitelist of vehicle classnames, vehicle classnames that the garage starts with, or other things formerly found in the config, it could make this module a lot more flexible.

It would also make the CBA options obsolete, and if you really want to get complicated with developing this, you might even be able to have multiple garages with different features scattered across the map. Maybe one garage only allows IDAP vehicles while another is military and won't allow civilian vehicles. Maybe you have a small garage of only 10 vehicles, and then a warehouse with 50 vehicles. Maybe you can exclude entire vehicle types like aircraft or boats. The possibilities from here get very substantial.

tl;dr start using the module itself to edit what it can do rather than CBA options

### Alternative soloutions

Extra modules that can be synced to the core one, and these modules would affect different features of the core module.

Kind of like how the "Add vehicles" module is already a thing.

**Comment by SpadesNeil** [comment]
Followup that relates to this issue https://github.com/HakonRydland/A3-HR-Garage/issues/8

If you add vehicle garage size in the module configs, you should do it like how ACE does logistics resources.

0 might be a valid choice for a mission maker where a garage starts unusable, but then gets more space as time goes on, so 0 should still be possible.
-1 can be read by the system as the default value. If the script read "garage size equals -1" it knows to check the CBA settings for what the value should be.
-10 or lower can be read by the system as being infinite. If the script reads "garage size equals less than -10" it knows to skip the capacity check entirely.

Just in case, you should have a sanity check that if the garage size equals a value less than zero but not one of those special values, it should revert to 0 and throw up an error in the console about what happened.

If you do this, issue #8 becomes obsolete.

**Comment by HakonRydland** [comment]
as most settings would be considered global and multiple garages would require a major rewrite that wont happen, still some stuff here sounds interesting and have some feasibility
